### PR TITLE
Update spec writing to sort keys and select values.

### LINKED ISSUE #122: Feature Request: Make the spec diff tests more resilient
While working on #115, the spec diff tests began failing (see below . After inspection most of the issues are related to the sort order for the required_links and the Xcode targets.

```
|| ==================== Test output for //test:xcodeproj_generator:
|| Spec doesn't match expected:
|| 25d24
|| <         "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/tools/generator/libgenerator.library.a",
|| 27c26,27
|| <         "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_pointfreeco_xctest_dynamic_overlay/libXCTestDynamicOverlay.a"
|| ---
|| >         "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_pointfreeco_xctest_dynamic_overlay/libXCTestDynamicOverlay.a",
|| >         "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/tools/generator/libgenerator.library.a"
|| 30c30
|| <         "@com_github_kylef_pathkit//:PathKit darwin_x86_64-fastbuild-ST-1b9bd654f600",
|| ---
|| >         "@com_github_tadija_aexml//:AEXML darwin_x86_64-fastbuild-ST-1b9bd654f600",
|| 72,73c72,73
|| <                 "PRODUCT_MODULE_NAME": "PathKit",
|| <                 "PRODUCT_NAME": "PathKit",
|| ---
|| >                 "PRODUCT_MODULE_NAME": "AEXML",
|| >                 "PRODUCT_NAME": "AEXML",
|| 83c83
|| <             "label": "@com_github_kylef_pathkit//:PathKit",
|| ---
|| >             "label": "@com_github_tadija_aexml//:AEXML",
|| 85c85
|| <             "name": "PathKit",
|| ---
|| >             "name": "AEXML",
|| 88,91c88,91
|| <                     "name": "PathKit.swiftmodule",
|| <                     "swiftdoc": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/PathKit.swiftdoc",
|| <                     "swiftmodule": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/PathKit.swiftmodule",
|| <                     "swiftsourceinfo": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/PathKit.swiftsourceinfo"
|| ---
|| >                     "name": "AEXML.swiftmodule",
|| >                     "swiftdoc": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_tadija_aexml/AEXML.swiftdoc",
|| >                     "swiftmodule": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_tadija_aexml/AEXML.swiftmodule",
|| >                     "swiftsourceinfo": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_tadija_aexml/AEXML.swiftsourceinfo"
|| 100,101c100,101
|| <                 "name": "PathKit",
|| <                 "path": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/libPathKit.a",
|| ---
|| >                 "name": "AEXML",
|| >                 "path": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_tadija_aexml/libAEXML.a",
|| 105c105,109
|| <                 "external/com_github_kylef_pathkit/Sources/PathKit.swift"
|| ---
|| >                 "external/com_github_tadija_aexml/Sources/AEXML/Document.swift",
|| >                 "external/com_github_tadija_aexml/Sources/AEXML/Element.swift",
|| >                 "external/com_github_tadija_aexml/Sources/AEXML/Error.swift",
|| >                 "external/com_github_tadija_aexml/Sources/AEXML/Options.swift",
|| >                 "external/com_github_tadija_aexml/Sources/AEXML/Parser.swift"
|| 109c113
|| <         "@com_github_tadija_aexml//:AEXML darwin_x86_64-fastbuild-ST-1b9bd654f600",
|| ---
|| >         "@com_github_kylef_pathkit//:PathKit darwin_x86_64-fastbuild-ST-1b9bd654f600",
|| 151,152c155,156
|| <                 "PRODUCT_MODULE_NAME": "AEXML",
|| <                 "PRODUCT_NAME": "AEXML",
|| ---
|| >                 "PRODUCT_MODULE_NAME": "PathKit",
|| >                 "PRODUCT_NAME": "PathKit",
|| 162c166
|| <             "label": "@com_github_tadija_aexml//:AEXML",
|| ---
|| >             "label": "@com_github_kylef_pathkit//:PathKit",
|| 164c168
|| <             "name": "AEXML",
|| ---
|| >             "name": "PathKit",
|| 167,170c171,174
|| <                     "name": "AEXML.swiftmodule",
|| <                     "swiftdoc": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_tadija_aexml/AEXML.swiftdoc",
|| <                     "swiftmodule": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_tadija_aexml/AEXML.swiftmodule",
|| <                     "swiftsourceinfo": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_tadija_aexml/AEXML.swiftsourceinfo"
|| ---
|| >                     "name": "PathKit.swiftmodule",
|| >                     "swiftdoc": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/PathKit.swiftdoc",
|| >                     "swiftmodule": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/PathKit.swiftmodule",
|| >                     "swiftsourceinfo": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/PathKit.swiftsourceinfo"
|| 179,180c183,184
|| <                 "name": "AEXML",
|| <                 "path": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_tadija_aexml/libAEXML.a",
|| ---
|| >                 "name": "PathKit",
|| >                 "path": "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/libPathKit.a",
|| 184,188c188
|| <                 "external/com_github_tadija_aexml/Sources/AEXML/Document.swift",
|| <                 "external/com_github_tadija_aexml/Sources/AEXML/Element.swift",
|| <                 "external/com_github_tadija_aexml/Sources/AEXML/Error.swift",
|| <                 "external/com_github_tadija_aexml/Sources/AEXML/Options.swift",
|| <                 "external/com_github_tadija_aexml/Sources/AEXML/Parser.swift"
|| ---
|| >                 "external/com_github_kylef_pathkit/Sources/PathKit.swift"
|| 421,422c421,422
|| <                 "@com_github_kylef_pathkit//:PathKit darwin_x86_64-fastbuild-ST-1b9bd654f600",
|| <                 "@com_github_tuist_xcodeproj//:XcodeProj darwin_x86_64-fastbuild-ST-1b9bd654f600"
|| ---
|| >                 "@com_github_tuist_xcodeproj//:XcodeProj darwin_x86_64-fastbuild-ST-1b9bd654f600",
|| >                 "@com_github_kylef_pathkit//:PathKit darwin_x86_64-fastbuild-ST-1b9bd654f600"
|| 784,785c784,785
|| <                 "//tools/generator:generator.library darwin_x86_64-fastbuild-ST-1b9bd654f600",
|| <                 "@com_github_pointfreeco_swift_custom_dump//:CustomDump darwin_x86_64-fastbuild-ST-1b9bd654f600"
|| ---
|| >                 "@com_github_pointfreeco_swift_custom_dump//:CustomDump darwin_x86_64-fastbuild-ST-1b9bd654f600",
|| >                 "//tools/generator:generator.library darwin_x86_64-fastbuild-ST-1b9bd654f600"
|| 882a883,884
|| >                 "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_pointfreeco_swift_custom_dump/libCustomDump.a",
|| >                 "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_pointfreeco_xctest_dynamic_overlay/libXCTestDynamicOverlay.a",
|| 886,888c888
|| <                 "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/libPathKit.a",
|| <                 "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_pointfreeco_swift_custom_dump/libCustomDump.a",
|| <                 "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_pointfreeco_xctest_dynamic_overlay/libXCTestDynamicOverlay.a"
|| ---
|| >                 "bazel-out/darwin_x86_64-fastbuild-ST-1b9bd654f600/bin/external/com_github_kylef_pathkit/libPathKit.a"
|| 
|| Run `bazel run //test/fixtures:update` if you wish to accept these changes.
|| ================================================================================

```

### PR DESCRIPTION
**Pull request by cgrindel**
Closes #122

- Added `flattened_key_values` library to support sorting of flattened key-values.
- Updated `xcodeproj` to sort Xcode targets, potential target merges, and required links.
- Updated the `validator.template.sh` to sort the JSON keys before performing the diff.

### CODE REVIEW

File: xcodeproj/internal/flattened_key_values.bzl [hunk 1]
```diff
@@ -0,0 +1,60 @@
+def _to_dict(iterable):
+    """Convert an iterable of flattened key-value pairs to a dictionary.
```
Comment (brentleyjones) [inline xcodeproj/internal/flattened_key_values.bzl:2]:
```suggestion
    """Converts an iterable of flattened key-value pairs to a dictionary.
```

File: xcodeproj/xcodeproj.bzl [hunk 2]
```diff
@@ -218,19 +226,19 @@ def make_xcodeproj_rule(*, transition = None):
...
             # only used by a local action, so it doesn't matter what it points
             # to.
             default = "@build_bazel_rules_apple//:LICENSE",
         ),
+        "_generator": attr.label(
+            cfg = "exec",
+            # TODO: Use universal generator when done debugging
+            default = Label("//tools/generator:generator"),
+            executable = True,
+        ),
```
Comment (brentleyjones) [inline xcodeproj/xcodeproj.bzl:241]: Small nit: can you submit these formatting changes as a separate PR?

**Review by brentleyjones — COMMENTED** [review]
Thanks for this!

File: xcodeproj/xcodeproj.bzl (hunk 2 above)
Comment (cgrindel) [reply xcodeproj/xcodeproj.bzl:241]: It wasn't intentional. My editor runs Buildifier on files that I touch. Let me see what I can do.

File: xcodeproj/internal/flattened_key_values.bzl (hunk 1 above)
Comment (cgrindel) [reply xcodeproj/internal/flattened_key_values.bzl]: Done

File: xcodeproj/xcodeproj.bzl (hunk 2 above)
Comment (cgrindel) [reply xcodeproj/xcodeproj.bzl]: Done.

**Review by brentleyjones — APPROVED** [review]

**Comment by brentleyjones** [comment]
Tests aren't running because of a known issue with PRs from forks. I'll run the tests locally and merge if they pass.
